### PR TITLE
refactor the model generation and unified it with enums and constants.

### PR DESCRIPTION
**Pull request by trusch**
refactor the model generation and unified it with enums and constants.

## How Has This Been Verified?
extended the tests and the existing ones are still passing

There are also no changes when this version of the generator is run against the hub spec.

## Checklist:

- [x] The change works as expected.
- [x] New code can be debugged via logs.
- [x] I have added tests to cover my changes.
- [x] I have locally run the tests and all new and existing tests passed.
- [ ] Requires updates to the documentation.
- [ ] I have made the required changes to the documents.

### CODE REVIEW

**Comment by LucasRoesler** [comment]
@trusch there is a lot of overlap with #6 can we merge that first? It will help reduce the scope of this PR and make it easier to review.

**Comment by trusch** [comment]
sure @LucasRoesler

File: pkg/generators/models/testdata/cases/enum/api.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+openapi: 3.0.0
+info:
+  version: 0.1.0
+  title: Test
+
+components:
+  schemas:
+    Foo:
```
Comment (LucasRoesler) [inline pkg/generators/models/testdata/cases/enum/api.yaml:8]: Let's make this easier to read 
```suggestion
    Name:
```

File: pkg/generators/models/testdata/cases/const/api.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+openapi: 3.0.0
+info:
+  version: 0.1.0
+  title: Test
+
+components:
+  schemas:
+    Foo:
```
Comment (LucasRoesler) [inline pkg/generators/models/testdata/cases/const/api.yaml:8]: Why new tests cases, there are already cases for 

`contants` https://github.com/contiamo/openapi-generator-go/tree/master/pkg/generators/models/testdata/cases/constants

and `enums` https://github.com/contiamo/openapi-generator-go/tree/master/pkg/generators/models/testdata/cases/enums

**Review by LucasRoesler — CHANGES_REQUESTED** [review]
It looks good, but let's remove the unused testdata files

**Review by LucasRoesler — APPROVED** [review]

**Pull request merged**